File: JavaScript/2091. Removing Minimum and Maximum From Array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minimumDeletions = function(nums) {
    var n = nums.length;
    var maxidx = 0;
    var minidx = 0;
    for(var i = 0; i < n; i++){
        if(nums[i] > nums[maxidx]){
            maxidx = i;
        }
        if(nums[i] < nums[minidx]){
            minidx = i;
        }
    }
    var fr  = Math.max(maxidx,minidx) + 1;
    var bc = n - Math.min(minidx,maxidx);
    var both = Math.min(minidx,maxidx) + 1 + (n - Math.max(minidx,maxidx));
    return Math.min(fr,Math.min(bc,both));
};

File: C#/2091. Removing Minimum and Maximum From Array.cs
public class Solution {
    public int MinimumDeletions(int[] nums) {
        int n = nums.Length;
        int maxidx = 0;
        int minidx = 0;
        for(int i = 0; i < n; i++){
            if(nums[i] > nums[maxidx]){
                maxidx = i;
            }
            if(nums[i] < nums[minidx]){
                minidx = i;
            }
        }
        int fr  = Math.Max(maxidx,minidx) + 1;
        int bc = n - Math.Min(minidx,maxidx);
        int both = Math.Min(minidx,maxidx) + 1 + (n - Math.Max(minidx,maxidx));
        return Math.Min(fr,Math.Min(bc,both));
    }
}
